src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -173,42 +188,121 @@ private bool ReadLineCore(bool consumeKeys)
                     {
                         return false;
                     }
+                    // Alt + B/F is what macOS will send in place of Alt + Left/Right
+                    else if (keyInfo.Modifiers == ConsoleModifiers.Alt && (OperatingSystem.IsMacOS() ? keyInfo.Key == ConsoleKey.B : keyInfo.Key == ConsoleKey.LeftArrow))
```

Comment:
I believe the correct keychord for non-macOS systems should be Ctrl+LeftKey/RightKey.